src/app/common/search-select/search-select.component.ts

```diff
@@ -9,28 +9,60 @@ import { SkillService } from 'src/app/skill.service';
   styleUrls: ['./search-select.component.scss']
 })
 export class SearchSelectComponent implements OnInit {
-  constructor(private skillService: SkillService) {}
+  constructor(private skillService: SkillService,
+    private eRef: ElementRef) {
+    this.allSkills = [];
+    this.skillExistsEmitter.emit(this.skillExists);
+  }
 
   selections = new FormControl();
   @Output() emitSelections = new EventEmitter<any>();
   title = 'Skills';
   options: Skill[];
   allSkills: Skill[];
+  @Output() skillExistsEmitter = new EventEmitter<any>();
+  skillExists = true;
+  @Output() skillTitleEmitter = new EventEmitter<any>();
+  searchTerm: string = "";
+
 
   selectOption() {
     this.emitSelections.emit(this.selections);
   }
 
+  // @HostListener('document:click', ['$event'])
+  // clickout(event) {
+  //   if(this.eRef.nativeElement.contains(event.target)) {
+  //     console.log('Clicked within search component');
+  //   } else {
+  //     console.log('Clicked outside search component');
+  //     //this.skillExists = true;
+  //     //this.skillExistsEmitter.emit(this.skillExists);
+  //   }
+  // }
+
   termChanged(event) {
+    this.searchTerm = event.target.value;
     this.options = this.allSkills.filter(el =>
-      el.title.includes(event.target.value)
+      el.title.toUpperCase().includes(this.searchTerm.toUpperCase())
     );
+    if(this.options.length == 0){
+      this.skillExists = false;
+      this.skillExistsEmitter.emit(this.skillExists);
+      this.skillTitleEmitter.emit(this.searchTerm);
+    } else {
+      this.skillExists = true;
+      this.skillExistsEmitter.emit(this.skillExists);
+    }
   }
 
   resetOptions() {
     this.options = this.allSkills;
+    
```

Comment:
Blank line not really needed here.

Reply:
And the extra blank lines below aren't really needed, although it isn't that significant.